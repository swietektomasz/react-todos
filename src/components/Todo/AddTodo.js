import React, { Component } from 'react'

export default class AddTodo extends Component {
  state = {
    text: '',
  }

  render() {
    return (
      <div>
        <input placeholder="add todo..." />
        <button onClick={this.props.handleAdding}>add</button>
      </div>
    )
  }
}
