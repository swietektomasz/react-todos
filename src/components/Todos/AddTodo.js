import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { addTodo } from '../../store/todos/actions'

export const AddTodo = props => {
  const [title, setTitle] = useState('')
  return (
    <div>
      <input
        placeholder="add todo..."
        defaultValue=""
        onChange={({ target: { value } }) => setTitle(value)}
      />
      <button onClick={() => props.addTodo(title)}>add</button>
    </div>
  )
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      addTodo,
    },
    dispatch,
  )
}

export default connect(
  null,
  mapDispatchToProps,
)(AddTodo)

AddTodo.propTypes = {
  addTodo: PropTypes.func.isRequired,
}
