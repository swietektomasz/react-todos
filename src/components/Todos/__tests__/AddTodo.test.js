import React from 'react'
import renderer from 'react-test-renderer'

import { AddTodo } from '../AddTodo'

const getWrapper = defaultProps => {
  const wrapper = renderer.create(<AddTodo {...defaultProps} />)

  return {
    wrapper,
    instance: wrapper.getInstance(),
    root: wrapper.root,
    json: wrapper.toJSON(),
  }
}

describe('<AddTodo />', () => {
  test('render', () => {
    const defaultProps = {
      addTodo: jest.fn(),
    }
    const { wrapper, root } = getWrapper(defaultProps)

    expect(wrapper).toMatchSnapshot()
    expect(root.findByType('input')).toBeDefined()
    expect(root.findByType('button')).toBeDefined()
  })

  test('input text', () => {
    const defaultProps = {
      addTodo: jest.fn(),
    }
    const { json, instance } = getWrapper(defaultProps)
    const insanceState = instance.state

    expect(json.children[0].props.value).toBe(insanceState.text)
  })

  test('calls addTodo on _handleSubmit', () => {
    const defaultProps = {
      addTodo: jest.fn(),
    }
    const { json, instance } = getWrapper(defaultProps)
    console.log(instance)
    const insanceState = instance.state
    instance._handleSubmit()
    expect(defaultProps.addTodo).toBeCalledWith('')

    instance._handleInputChange({ target: { value: '123' } })
    instance._handleSubmit()
    expect(defaultProps.addTodo).toBeCalledWith('123')

    expect(json.children[0].props.value).toBe(insanceState.text)
  })
})
