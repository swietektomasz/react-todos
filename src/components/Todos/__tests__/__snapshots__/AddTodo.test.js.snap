exports[`<AddTodo /> render 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="add todo..."
    value="" />
  <button
    onClick={[Function]}>
    add
  </button>
</div>
`;
